add "Delete all cookies" to privacy settings

balcy opened the pull request:
implements basic feature request of https://github.com/ubports/morph-browser/issues/163

advocatux committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (222 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/es/

cibersheep committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (222 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/ca/

Flohack74 committed:
Translated using Weblate (German)

Currently translated at 73.4% (163 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/de/

ivoxavier committed:
Translated using Weblate (Portuguese)

Currently translated at 100.0% (222 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/pt/

Daniel20000522 committed:
Translated using Weblate (Polish)

Currently translated at 100.0% (222 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/pl/

Commit:
Translated using Weblate (Norwegian Bokmål)

Currently translated at 73.9% (164 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/nb/

comradekingu committed:
Translated using Weblate (Norwegian Bokmål)

Currently translated at 88.3% (196 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/nb/

weblate committed:
Merge branch 'origin/xenial' into Weblate.

adrmzz committed:
Translated using Weblate (Sardinian)

Currently translated at 100.0% (222 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/sc/

Commit:
Translated using Weblate (French)

Currently translated at 79.3% (176 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/fr/

MK1973 committed:
Translated using Weblate (Czech)

Currently translated at 100.0% (222 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/cs/

RedXXIII committed:
Translated using Weblate (French)

Currently translated at 79.3% (176 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/fr/

Commit:
Translated using Weblate (French)

Currently translated at 79.3% (176 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/fr/

Commit:
Translated using Weblate (French)

Currently translated at 79.7% (177 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/fr/

weblate committed:
Merge branch 'origin/xenial' into Weblate.

HendriXXX committed:
Translated using Weblate (Finnish)

Currently translated at 72.5% (161 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/fi/

Atalanttore committed:
Translated using Weblate (German)

Currently translated at 88.3% (196 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/de/

Commit:
Translated using Weblate (French)

Currently translated at 100.0% (222 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/fr/

RedXXIII committed:
Translated using Weblate (French)

Currently translated at 100.0% (222 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/fr/

balcy committed:
add "Delete all cookies" to privacy settings

balcy committed:
add "Delete all cookies" to privacy settings

Commit:
Translated using Weblate (Dutch)

Currently translated at 76.1% (169 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/nl/

Commit:
Translated using Weblate (Dutch)

Currently translated at 100.0% (222 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/nl/

balcy committed:
remove property incognito, because it is already present in Morph/Web/MorphWebView.qml

weblate committed:
Merge branch 'origin/xenial' into Weblate.

balcy committed:
Merge pull request #235 from ubports-weblate/weblate-ubports-morph-browser

Update from Weblate

weblate committed:
Merge branch 'origin/xenial' into Weblate.

balcy committed:
create a sharedIncognitoContext (offTheRecord = true) that is used for incognito webviews
(all non-incognito webviews use the profile "SharedWebContext.sharedContext", the incognito webviews use the profile "SharedWebContext.sharedIncognitoContext")

TheRealDonhammer committed:
Translated using Weblate (Danish)

Currently translated at 71.6% (159 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/da/

balcy committed:
Merge pull request #245 from ubports-weblate/weblate-ubports-morph-browser

Update from Weblate

HendriXXX committed:
Translated using Weblate (Finnish)

Currently translated at 77.5% (172 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/fi/

comradekingu committed:
Translated using Weblate (Norwegian Bokmål)

Currently translated at 92.3% (205 of 222 strings)

Translation: UBports/Morph Browser
Translate-URL: https://translate.ubports.com/projects/ubports/morph-browser/nb/

UniversalSuperBox reviewed (changes requested):
I think that [QWebEngineCookieStore::deleteAllCookies()](https://doc.qt.io/archives/qt-5.11/qwebenginecookiestore.html#deleteAllCookies) would be a much better option to use here. [QQuickWebEngineProfile::cookieStore()](https://doc.qt.io/archives/qt-5.11/qquickwebengineprofile.html#cookieStore) or [QWebEngineProfile::cookieStore()](https://doc.qt.io/archives/qt-5.11/qwebengineprofile.html#cookieStore) can be used to access the cookie store.

Aury88 committed:
Remove the t=canonical string from DDG search (#251)

Revert commit 04f61fcc88d06137525c7866fc0e5029dffab266 that added the t=canonical string to the searches

balcy committed:
Merge pull request #248 from ubports-weblate/weblate-ubports-morph-browser

Update from Weblate

balcy committed:
rebase with xenial
Merge branch 'xenial_-_deletecookies' of https://github.com/ubports/morph-browser into xenial_-_deletecookies

balcy committed:
use deleteAllCookies method from QtWebEngine to delete the cookies (Settings -> Privacy & Permissions -> Clear all cookies)
(they are deleted from both the standard and the incognito profile)

delete the http cache from the standard and incognito profile (Settings -> Privacy & Permissions -> Clear Cache)

balcy commented:
hmm strange it worked with clickable... have to look into it

balcy committed:
add QtWebEngine (dev) dependencies

balcy commented:
or it built successfully now. The reason for the rebase was, that I wanted to bring in the new "SharedIncognitoContext" from https://github.com/ubports/morph-browser/pull/234/commits/07572e028e575892ff40a7d285b7214fbfd3500e.
the cookies from the standard profile, and the incognito profile are both deleted.

balcy committed:
rename deleteAllCookies() -> deleteAllCookiesOfProfile()

UniversalSuperBox commented on src/app/webcontainer/WebappSettingsPage.qml:
```diff
@@ -202,6 +203,17 @@ FocusScope {
+                        ListItem {
+                            objectName: "privacy.clearAllCookies"
+                            ListItemLayout {
+                                title.text: i18n.tr("Clear all cookies")
```
This option is sentence cased but the two above it are Title Cased. Shall we pick one or the other?

balcy replied:
~hmm, is "Clear All Cookies" ok ?~

balcy replied:
On the other hand: the first 4 items are not Title Cased (Only allow ..., Domain blacklist/whitelist, Domain specific settings, Camera & microphone)
maybe it is easier to make it "Clear browsing history" and "Clear cache"
It is used in the question for the user, too.
So I would vote for this.

balcy committed:
adapt casing for "Clear browsing history" and "Clear cache"

balcy committed:
adapt casing of "Clear cache" for webapp settings

balcy committed:
adapt Jenkinsfile

balcy commented:
hmm the build problem is strange (adapting the Jenkinsfile seemed to have worked on branch https://github.com/ubports/morph-browser/pull/234

I think we might be able to merge it to xenial like that one: 
https://github.com/ubports/morph-browser/pull/256

UniversalSuperBox commented:
@balcy, it appears that commits from the xenial branch have appeared behind your commits in this branch. Could you please re-create this PR or push the branch so that only your changes on top of the xenial branch are here?
